JDBC SNAPSHOT Jars

hannes opened the pull request:
This PR fully automates the JDBC/Maven release process. We also start uploading SNAPSHOT releases to Maven central. I've centralised the JDBC builds into their own separate workflow as well.

hannes committed:
attempt 1

hannes committed:
spaces

hannes committed:
spaaace

hannes committed:
should do something

hannes committed:
grr

hannes committed:
combining something

hannes committed:
Merge branch 'master' of github.com:duckdb/duckdb

hannes committed:
this may work

hannes committed:
spacing yay

hannes committed:
sure

hannes committed:
quotes?

hannes committed:
grrr

hannes committed:
moar pipes

hannes committed:
Merge branch 'master' of github.com:duckdb/duckdb

hannes committed:
make cleanMerge branch 'master' of github.com:duckdb/duckdb # Please enter
a commit message to explain why this merge is necessary, # especially if
it merges an updated upstream into a topic branch. # # Lines starting with
'#' will be ignored, and an empty message aborts # the commit.

hannes committed:
maven fun

hannes committed:
hey it does something

hannes committed:
some leftovers

hannes commented:
Not so sure about the `paths-ignore:` there

hannes committed:
not trying this on prs

hannes committed:
spaces

Mytherin reviewed (approved):
Thanks! Looks good to me. Good idea to split the Java build into a separate CI workflow.

hannes committed:
make cleanMerge branch 'master' of github.com:duckdb/duckdb # Please enter
a commit message to explain why this merge is necessary, # especially if
it merges an updated upstream into a topic branch. # # Lines starting with
'#' will be ignored, and an empty message aborts # the commit.
amke debug

georgewfraser commented:
Thank you!

samansmink commented on scripts/jdbc_maven_deploy.py:
```diff
@@ -108,79 +128,70 @@ def exec(cmd):
 
 # run basic tests, it should now work on whatever platform this is
-exec("java -cp %s org.duckdb.test.TestDuckDBJDBC" % binary_jar)
+# exec("java -cp %s org.duckdb.test.TestDuckDBJDBC" % binary_jar)
```
should this remain commented out?

hannes replied:
good catch, let me try

hannes committed:
re-enable tests
